pkg/identity/identity_test.go

```diff
@@ -184,20 +184,53 @@ var _ = Describe("GetCredentials", func() {
 			Expect(err).To(HaveOccurred())
 		})
 	})
+})
 
-	Context("prerequisites missing", func() {
-		It("should error if cluster is missing", func() {
-			_, err := GetCredentials(ctx, k8sclient, nil, manager.DefaultPodNamespace)
+var _ = Describe("validateInputs", func() {
+	var (
+		ns      *corev1.Namespace
+		cluster *infrav1.VSphereCluster
+	)
+
+	BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "namespace-",
+			},
+		}
+		Expect(k8sclient.Create(ctx, ns)).To(Succeed())
+
+		cluster = &infrav1.VSphereCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "cluster-",
+				Namespace:    ns.Name,
+			},
+		}
+		Expect(k8sclient.Create(ctx, cluster)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		Expect(k8sclient.Delete(ctx, ns)).To(Succeed())
+		Expect(k8sclient.Delete(ctx, cluster)).To(Succeed())
```

Comment:
Reverse the order, though I think, just deleting the namespace should delete the objects contained in the namespace automaGically.